### PR TITLE
Adding a "remove subs from a host collection" step

### PR DESCRIPTION
**Pull request by swadeley**
Hello

this depends on:
https://github.com/SatelliteQE/nailgun/pull/747

### CODE REVIEW

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/api/test_hostcollection.py -k test_positive_add_remove_subscription
================================== test session starts =============================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: xdist-2.1.0, services-1.3.1, mock-1.10.4, forked-1.3.0, cov-2.10.1
collecting ... 2020-09-09 15:34:11 - conftest - DEBUG - Collected 20 test cases
collected 20 items / 19 deselected / 1 selected                                                                  

tests/foreman/api/test_hostcollection.py .                                                                 [100%]


================ 1 passed, 19 deselected, 2 warnings in 199.87 seconds ==============================

File: tests/foreman/api/test_hostcollection.py [hunk 1]
```diff
@@ -448,22 +454,25 @@ def test_positive_add_subscription(self):
...
+            for host_id in host_ids:
+                req = request(
+                    'GET',
+                    '{0}/api/v2/hosts/{1}/subscriptions'.format(
+                        settings.server.get_url(), host_id
+                    ),
+                    verify=False,
+                    auth=settings.server.get_credentials(),
+                    headers={'content-type': 'application/json'},
+                )
```
Comment (mirekdlugosz) [inline tests/foreman/api/test_hostcollection.py:477]: I vaguely feel we had this conversation before, but that really should be part of Nailgun.

**Review by mirekdlugosz — APPROVED** [review]
ACK, but that GET to /hosts/:host/subscriptions really should be part of Nailgun

File: tests/foreman/api/test_hostcollection.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/api/test_hostcollection.py:477]: Yes, we have:
https://github.com/SatelliteQE/nailgun/issues/750

**Review by pondrejk — APPROVED** [review]

File: tests/foreman/api/test_hostcollection.py [hunk 2]
```diff
@@ -392,17 +391,22 @@ def test_negative_create_with_invalid_name(self):
...
 
         :steps:
 
-            1. Create a new or use an existing subscription
-            2. Add the subscription to the members of the host collection
+            1. Create AK and HC, add AK to HC
+            2. Create product so we can use it's subscription
+            3. Create some VMs and register them with AK so they are in HC
+            2. Add the subscription to the members of the Host Collection
+            3. Assert subscription is added
```
Comment (pondrejk) [inline tests/foreman/api/test_hostcollection.py:405]: the numbering here probably makes make_docstrings fail in travis

Comment (swadeley) [reply tests/foreman/api/test_hostcollection.py]: oh dear! fix made, thank you for spotting that @pondrejk

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/api/test_hostcollection.py -k test_positive_add_remove_subscription
========================== test session starts ===================================
platform linux -- Python 3.7.9, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: xdist-2.1.0, services-1.3.1, mock-1.10.4, forked-1.3.0, cov-2.10.1
collecting ... 2020-09-22 06:04:03 - conftest - DEBUG - Collected 20 test cases
collected 20 items / 19 deselected / 1 selected                                                                  

tests/foreman/api/test_hostcollection.py .                                                                 [100%]

============= 1 passed, 19 deselected, 2 warnings in 198.37 seconds =================

**Comment by swadeley** [comment]
Thank you

**Pull request merged**